src/Adapter/Order/CommandHandler/AddOrderFromBackOfficeHandler.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Order\CommandHandler;
+
+use Address;
+use BoOrderCore;
+use Cart;
+use Configuration;
+use Context;
+use Currency;
+use Customer;
+use Employee;
+use Exception;
+use Module;
+use PrestaShop\PrestaShop\Core\Domain\Order\Command\AddOrderFromBackOfficeCommand;
+use PrestaShop\PrestaShop\Core\Domain\Order\CommandHandler\AddOrderFromBackOfficeHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+
+/**
+ * @internal
+ */
+final class AddOrderFromBackOfficeHandler implements AddOrderFromBackOfficeHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(AddOrderFromBackOfficeCommand $command)
+    {
+        $paymentModule = !Configuration::get('PS_CATALOG_MODE') ?
+            Module::getInstanceByName($command->getPaymentModuleName()) :
```

Comment:
Explosion. :rocket: :boom:

I think this is an edge case, I wonder if that ever happened to someone. Should we validate this case here?

Reply:
If we can just check what `Module::getInstanceByName($command->getPaymentModuleName())` does in this case:
- if it throws an Exception, encapsulate it into a domain exception "payment module not available"
- if it returns null, throw a domain exception

Although it's an edge case, it's a lot more dangerous that other edge cases because it involves external software collaboration.

Reply:
Done.